### PR TITLE
API: Update Blocks-in-range endpoint Attempt 2 [ECR-791]

### PR DESCRIPTION
**Pull request by quicky84**
This PR is directly related to #351: appends suggested changes, optimizes the code, and adds requested test.

### CODE REVIEW

File: exonum/src/explorer/mod.rs [hunk 1]
```diff
@@ -76,6 +76,24 @@ pub enum TxStatus {
     },
 }
 
+/// Range information. Range borders are included.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Range {
```
Comment (vitvakatu) [inline exonum/src/explorer/mod.rs]: Can't we use `std::Range`?

Comment (quicky84) [reply exonum/src/explorer/mod.rs]: https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html this one?

Comment (vitvakatu) [reply exonum/src/explorer/mod.rs]: https://doc.rust-lang.org/std/ops/struct.Range.html this? Or you need inclusive one?

Comment (quicky84) [reply exonum/src/explorer/mod.rs]: By the logic of the block exploration, it makes sense to work with closed intervals.

Comment (vitvakatu) [reply exonum/src/explorer/mod.rs]: Sadly, I don't think it's possible as it's nightly-only API atm.

**Comment by vitvakatu** [comment]
Replaced by #509

**Pull request closed without merging**